guide.md

```diff
@@ -28,23 +57,31 @@ I believe this lesson has value for those who are a bit more experienced but we
 probably need to rethink this lesson to make it more approachable and
 interesting for all participants.
 
+In this lesson we have iterated between showing and discussing explicit
+examples, which can be boring for those who are not familiar with the language
+of the example, and a general overview (current version), where the feedback
+can be that this is not explicit enough. To accommodate the many languages we
+have chosen to maximize group work where code complexity, modular design, and
+side effects are discussed in the context of a particular programming language.
 
-## Intended learning outcomes
+We have tried various formats:
+- Slides and discussion (this repository in the past)
+- Fewer slides, group work, and discussion (this reposistory now)
+- Demo with discussions and suggestions via collaborative notes (https://github.com/coderefinery/modular-type-along)
 
-By the end of this lesson, learners should:
-- know the concepts of side effects, state, and purity
-- be able to motivate the use of pure functions
-- be able to visualize their code in form of a call tree (note that this is currently not in the material)
-- know where in the call tree to locate "impurity" (state, side effects, input/output) and which part of the call tree
-  to keep pure
-- be able to recognize when to divide a code portion into a function
-- be able to recognize when to divide a function into functions
-- understand the advantages (and disadvantages) of encapsulation
-- be aware that code complexity might limit reuse and introduce bugs
-- know that implementation choices are often a balance between efficiency in terms of human time, CPU time, memory, disk, ...
 
+## Timing
 
-## How to teach this lesson
+We have tried this lesson with 1-1.5 hours but perhaps this is too short.
+
+It is difficult to not become too abstract or too specific on one language.
+
+Perhaps we should give this lesson more time and perhaps we should try a "bring
+your own script/code" format where we improve our projects in a group work
+setting?
```

Comment:
Nice idea, but I guess that might end up too difficult for some and definitely taking too long.  Would be nice to do when we have more time as part of a hackathon of some sort (or RSE stuff).